[bitnami/cassandra] Making TLS existingSecrets more configurable

tbalzer opened the pull request:
<!--
 Before you open the request please review the following guidelines and tips to help it be more easily integrated:

 - Describe the scope of your change - i.e. what the change does.
 - Describe any known limitations with your change.
 - Please run any tests or examples that can exercise your modified code.

 Thank you for contributing! We will try to test and integrate the change as soon as we can, but be aware we have many GitHub repositories to manage and can't immediately respond to every request. There is no need to bump or check in on a pull request (it will clutter the discussion of the request).

 Also don't be worried if the request is closed or not integrated sometimes the priorities of Bitnami might not match the priorities of the pull request. Don't fret, the open source community thrives on forks and GitHub makes it easy to keep your changes in a forked repo.
 -->

### Description of the change

Added values to allow for more configuration of the TLS settings, e.g. alternate secret keys, split secrets.
<!-- Describe the scope of your change - i.e. what the change does. -->

### Benefits

E.g. interoperability with cert-manager and trust-manager.
<!-- What benefits will be realized by the code change? -->

### Possible drawbacks

None really?
<!-- Describe any known limitations with your change -->

### Applicable issues

<!-- Enter any applicable Issues here (You can reference an issue using #) -->
- fixes #28687 

### Additional information

<!-- If there's anything else that's important and relevant to your pull request, mention that information here.-->
TLS configuration is not currently covered by automated tests.

### Checklist

<!-- [Place an '[X]' (no spaces) in all applicable fields. Please remove unrelated fields.] -->

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

tbalzer commented:
@carrodher I'd appreciate any help you can give me with getting the pipeline to succeed.

I have invited @bitnami-bot to the repository which could be accepted to resolve this.
As it's an organization owned fork I can't use the GitHub builtin-functionality to give access (as it's only for personally owned forks).

Unfortunately the log of the changelog job doesn't show the diff which I could apply and I seem to be unable to locally reproduce what this job generates.

migruiz4 commented on bitnami/cassandra/values.yaml:
```diff
@@ -940,18 +941,28 @@ metrics:
   clientEncryption: false
   autoGenerated: false
   existingSecret: ""
+  existingTrustSecret: ""
```
For consistency, would it be possible to rename this as `existingTruststoreSecret`?

tbalzer replied:
Yes, good idea. Done.

migruiz4 reviewed (commented):
Thank you very much for your contribution @tbalzer! Could you please take a look at my comments?

tbalzer commented on bitnami/cassandra/templates/statefulset.yaml:
```diff
@@ -552,9 +552,26 @@ spec:
         {{- if (include "cassandra.tlsEncryption" . ) }}
         - name: certs
+        {{- if (include "cassandra.createTlsSecret" . ) }}
           secret:
-            secretName: {{ include "cassandra.tlsSecretName" . }}
+            secretName: {{ include "cassandra.keystoreSecretName" . }}
             defaultMode: 256
+        {{- else }}
+          projected:
+            sources:
+              - secret:
+                  name: {{ include "cassandra.truststoreSecretName" . }}
+                  items:
+                    - key: {{ .Values.tls.secretKeys.truststore }}
+                      path: truststore
+                      mode: 256
+              - secret:
+                  name: {{ include "cassandra.keystoreSecretName" . }}
+                  items:
+                    - key: {{ .Values.tls.secretKeys.keystore }}
+                      path: keystore
+                      mode: 256
+        {{- end }}
```
The created secret from https://github.com/bitnami/charts/blob/main/bitnami/cassandra/templates/tls-secret.yaml still uses hardcoded keys, as those are then not in JKS format. This is then handled differently in the init-container.

I think it wouldn't make sense there to use a projected secret then?

tbalzer replied:
@migruiz4 It's been a month now, any thoughts?

tbalzer commented:
> Thank you very much for your contribution @tbalzer! Could you please take a look at my comments?

Thanks @migruiz4, I've incorporated all but one and left a reply the last. When you have time please take a look.

tbalzer commented:
Hey @migruiz4 , this is getting stale right now. Anything I can do to move this forward?

tbalzer commented:
> Sorry for the delay! Please note a new version was released in the meantime, you need to rebase from main and bump the chart version in the _Chart.yaml_ again

I've rebased and updated accordingly.

tbalzer commented:
@carrodher @migruiz4 Can I get a review here?
